TM1py/Services/CellService.py

```diff
@@ -991,6 +1008,138 @@ def write_through_unbound_process(self, cube_name: str, cellset_as_dict: Dict, i
+        variable_list = []
+        for n, dimension in enumerate(cube_dimensions):
+            variable_list.append(variable_prefix + dimension.replace(' ', ''))
+            dataload_process.add_variable(name=variable_list[n], variable_type='String')
+
+        df = build_pandas_dataframe_from_cellset(cellset_as_dict)
+        # Create a comma separated list of variables
+        comma_sep_var_elements = ",".join(variable_list)
+
+        # Add value variable, type String to make possible to load both Numeric and String
+        variable_value = 'vValue'
+        dataload_process.add_variable(name=variable_value, variable_type='String')
+
+        # Write the statement for Cell Is Updateable
+        if skip_non_updateable:
+            cell_is_updateable_pre = f"If( CellIsUpdateable('{cube_name}',{comma_sep_var_elements}) = 1 );"
+            cell_is_updateable_post = '\rElse;\r' \
+                                      '   ItemSkip;\r' \
+                                      'EndIf;\r'
+        else:
+            cell_is_updateable_pre = ""
+            cell_is_updateable_post = ""
+
+        # Define TI write function based on parameter 'increment'
+        function_str = "CellIncrementN" if increment else "CellPutN"
+
+        # Define input statement depending on measure element's type: numeric or string
+        input_statement = f"""
+        If( ElementType('{cube_measure}', '', {variable_cube_measure}) @= 'N' );
+
+                nValue = StringToNumber({variable_value});
+                {function_str}(nValue,'{cube_name}',{comma_sep_var_elements});
+
+        ElseIf( ElementType('{cube_measure}', '', {variable_cube_measure}) @= 'S' );
+
+                sValue = {variable_value};
+                CellPutS(sValue,'{cube_name}',{comma_sep_var_elements});
+
+        EndIf;"""
+
+        # Define statement for Data section
+        data_statement = cell_is_updateable_pre + input_statement + cell_is_updateable_post
+        dataload_process.data_procedure = data_statement
+
+        # Define active sandbox function on Prolog section
+        enable_sandbox = self.generate_enable_sandbox_ti(sandbox_name)
+        dataload_process.prolog_procedure = '\r' + enable_sandbox
+
+        # Call the TI process with result
+        success, status, log_file = process_service.execute_process_with_return(process=dataload_process)
+        if not success:
+            if status == 'HasMinorErrors':
+                raise TM1pyWritePartialFailureException(status, log_file, success)
+            else:
+                raise TM1pyWriteFailureException(status, log_file)
+
+        # Delete CSV file on TM1 server
+        application_service.delete(path='TM1py', application_type='DOCUMENT', application_name=export_filename)
+        # Optinally delete the folder TM1py
+        # application_service.delete(path='', application_type='FOLDER', application_name='TM1py')
+
+        # Delete CSV file local
+        os.remove(export_path_to_file)
```

Comment:
We should also add a parameter to clean up the applications folder

Reply:
Let's discuss this in #882
Yeah. I think we could make this optional. If the function is called in parallel though, it would be a nuisance if a separate thread, process, or application removed the tm1py folder in though...